### PR TITLE
[libdatachannel] enable usage of feature srtp provide by vcpkg

### LINKED ISSUE #20721: [libdatachannel] cmake failure
It would seem that libdatachannel fails to configure if the `srtp` feature is enabled, due to a missing CMake target.

**Host Environment**
- OS: Linux
- Compiler: clang 11.0.1

**To Reproduce**
+ `CMakeLists.txt`
   ```cmake
	cmake_minimum_required(VERSION 3.18)

	# Bootstrap vcpkg
	set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake CACHE STRING "Vcpkg toolchain file")

	project(myproj)

	set(CMAKE_CXX_STANDARD 17)

	find_package(LibDataChannel CONFIG REQUIRED)

	add_executable(main main.cpp)

	target_link_libraries(server PRIVATE LibDataChannel::LibDataChannel)
   ```
+ `vcpkg.json`
   ```json
   {
		"dependencies": [
			{
				"name": "libdatachannel",
				"features": [
					"srtp"
				]
			}
		]
	}
   ```

**Failure logs** 
- ```log
	[main] Configuring folder: myproject 
	[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/clang-11 -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/clang++-11 -H/workspaces/myproject -B/workspaces/myproject/build -G "Ninja Multi-Config"
	[cmake] Not searching for unused variables given on the command line.
	[cmake] -- Running vcpkg install
	[cmake] Detecting compiler hash for triplet x64-linux...
	[cmake] All requested packages are currently installed.
	[cmake] Restored 0 packages from /home/vscode/.cache/vcpkg/archives in 652 ns. Use --debug to see more details.
	[cmake] 
	[cmake] Total elapsed time: 5.916 us
	[cmake] 
	[cmake] The package libdatachannel provides CMake targets:
	[cmake] 
	[cmake]     find_package(LibDataChannel CONFIG REQUIRED)
	[cmake]     target_link_libraries(main PRIVATE LibDataChannel::datachannel LibDataChannel::LibDataChannel)
	[cmake] 
	[cmake] -- Running vcpkg install - done
	[cmake] -- Configuring done
	[cmake] CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:538 (_add_executable):
	[cmake]   Target "server" links to target "SRTP::SRTP" but the target was not found.
	[cmake]   Perhaps a find_package() call is missing for an IMPORTED target, or an
	[cmake]   ALIAS target is missing?
	[cmake] Call Stack (most recent call first):
	[cmake]   server/CMakeLists.txt:3 (add_executable)
	[cmake] 
	[cmake] 
	[cmake] CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:538 (_add_executable):
	[cmake]   Target "server" links to target "SRTP::SRTP" but the target was not found.
	[cmake]   Perhaps a find_package() call is missing for an IMPORTED target, or an
	[cmake]   ALIAS target is missing?
	[cmake] Call Stack (most recent call first):
	[cmake]   server/CMakeLists.txt:3 (add_executable)
	[cmake] 
	[cmake] 
	[cmake] CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:538 (_add_executable):
	[cmake]   Target "server" links to target "SRTP::SRTP" but the target was not found.
	[cmake]   Perhaps a find_package() call is missing for an IMPORTED target, or an
	[cmake]   ALIAS target is missing?
	[cmake] Call Stack (most recent call first):
	[cmake]   server/CMakeLists.txt:3 (add_executable)
	[cmake] 
	[cmake] 
	[cmake] -- Generating done
	[cmake] CMake Generate step failed.  Build files cannot be regenerated correctly.
  ```

**Additional context**
This issue seems to stem partially from the fact that libsrtp doesn't actually emit any CMake exports, whereas libdatachannel appears to look for them under a certain name. I've submitted two pull requests to resolve this issue.

See:
- #20720
- #20722

### PR DESCRIPTION
**Pull request by FrankXie05**
depend on #20722
Fix #20721
`libdatachannel` uses the cmake configuration exported by `libsrtp`.
````
CMake Error at /home/test/vcpkg/scripts/buildsystems/vcpkg.cmake:582 (_add_executable):
  Target "main" links to target "libSRTP::srtp2" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:7 (add_executable)
````

### CODE REVIEW

File: ports/libdatachannel/0002-fix-for-srtp.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ 	if(USE_SYSTEM_SRTP)
+-		find_package(libSRTP REQUIRED)
+-		if(NOT TARGET libSRTP::srtp2)
+-			add_library(libSRTP::srtp2 UNKNOWN IMPORTED)
+-			set_target_properties(libSRTP::srtp2 PROPERTIES
+-				INTERFACE_INCLUDE_DIRECTORIES ${SRTP_INCLUDE_DIRS}
+-				IMPORTED_LINK_INTERFACE_LANGUAGES C
+-				IMPORTED_LOCATION ${SRTP_LIBRARIES})
+-		endif()
++		find_package(libsrtp CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/libdatachannel/0002-fix-for-srtp.patch:17]: This is not broken. `find_package(libSRTP)` will happily use vcpkg's `share/libsrtp/libSRTPConfig.cmake`.

**Review by dg0yt — CHANGES_REQUESTED** [review]
IIUC what is needed is `find_dependency(libSRTP)` in libdatachannels exported CMake config.

File: versions/l-/libdatachannel.json [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
-    "version-semver": "0.17.10",
-    "port-version": 0
+      "git-tree": "c84a9afc5c8c47aed97ebcbb22601a3b0394b27c",
+      "version-semver": "0.17.10",
+      "port-version": 1
+    },
+    {
+      "git-tree": "8a0b138705a4ef7e8b7117c8674d018bf281335f",
+      "version-semver": "0.17.10",
+      "port-version": 0
```
Comment (JonLiu1993) [inline versions/l-/libdatachannel.json:11]: @FrankXie05, Please don't change the history `git-tree`.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/libdatachannel/portfile.cmake [hunk 3]
```diff
@@ -47,4 +48,5 @@ ${DATACHANNEL_CONFIG}")
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
 
+# License
```
Comment (dg0yt) [inline ports/libdatachannel/portfile.cmake:51]: Can we *not* add this comment? It has no added value.

File: ports/libdatachannel/0002-fix-for-srtp.patch (hunk 1 above)
Comment (dg0yt) [reply ports/libdatachannel/0002-fix-for-srtp.patch:17]: Ping.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
The usage of port `llibsrtp` shuold be fixed. Otherwise, it won't work. 
Re-draft this Pr until usage is resolved.
````
find_package(libSRTP CONFIG REQUIRED)
target_link_libraries(usage PRIVATE libSRTP::srtp2)
````

````
1> [CMake] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
1> [CMake] CMake Error at E:/port-bug/vcpkg/installed/x64-windows/share/libsrtp/libSRTPTargets.cmake:85 (message):
1> [CMake]   The imported target "libSRTP::srtp2" references the file
1> [CMake] 
1> [CMake]      "E:/port-bug/vcpkg/installed/x64-windows/debug/lib/srtp2.dll"
1> [CMake] 
1> [CMake]   but this file does not exist.  Possible reasons include:
1> [CMake] 
1> [CMake]   * The file was deleted, renamed, or moved to another location.
1> [CMake] 
1> [CMake]   * An install or uninstall procedure did not complete successfully.
1> [CMake] 
1> [CMake]   * The installation package was faulty and contained
1> [CMake] 
1> [CMake]      "E:/port-bug/vcpkg/installed/x64-windows/share/libsrtp/libSRTPTargets.cmake"
1> [CMake] 
1> [CMake]   but not all the files it references.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   E:/port-bug/vcpkg/installed/x64-windows/share/libsrtp/libSRTPConfig.cmake:10 (include)
1> [CMake]   E:/port-bug/vcpkg/scripts/buildsystems/vcpkg.cmake:834 (_find_package)
1> [CMake]   usage/CMakeLists.txt:11 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> [CMake] See also "E:/usage/out/build/x64-Debug/CMakeFiles/CMakeOutput.log".
1> [CMake] See also "E:/usage/out/build/x64-Debug/CMakeFiles/CMakeError.log"
````

**Review by dg0yt — CHANGES_REQUESTED** [review]
In the manifest, `srtp` is presented as a feature. But AFAIU srtp is not a feature but a core dependency, to avoid a vendored copy. Or maybe a dependency of a to-be-added feature `media`, to control `NO_MEDIA`.

**Comment by FrankXie05** [comment]
Duplicated of #27540

**Pull request closed without merging**